CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,187 @@
+
+An exchange can have the following states:
+1. Proposed - The exchange has been proposed, but not yet approved by the approver. The proposer may still cancel their proposal and be refunded their deposit.
+2. Approved - The exchange has been proposed and approved by the approver, but not yet executed or vetoed. The proposer may no longer cancel their proposal at this point.
+3. Executed - The exchange has been proposed, approved by the approver, and executed.
+4. Cancelled - The exchange has been cancelled by the proposer when it was in the Proposed state, or it has been vetoed by Governance when it was in the Approved stage.
+
+### Specific Changes:
+
+#### Modifications of existing contracts:
+
+##### `StableToken.sol`
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
+1. **Modification to [mint()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L224)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted senders, Exchange and `Validators.sol`.
+2. **Modification to [burn()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L273)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted sender, the Exchange.
+
+#### New contracts:
+
+##### `GrandaMento.sol`
+
+A new contract, `GrandaMento.sol`, is created, added to the Registry with the identifier `GrandaMento`, and owned by Governance. `GrandaMento` is a singleton contract that is capable of facilitating exchanges for multiple stable tokens-- that is, with the approval of Governance, `GrandaMento` could support `CELO <-> cUSD`, `CELO <-> cEUR`, and future `CELO <-> cXXX` exchanges. `GrandaMento` must be a spender of `Reserve.sol`, and supported stable tokens must allow `GrandaMento` to mint/burn them.
+
+The contract has the following configurable parameters:
+
+1. **`address public approver;`** - Set by Governance. Intended to be a multisig and has the authority to approve exchanges. The signers for the multisig are at the discretion of Governance, but it would comprise of community members or members on behalf of organizations that are aligned with the Celo network/ecosystem.
+2. **`uint256 public exchangeWaitPeriodSeconds;`** - Set by Governance. The minimum amount of time in seconds that must elapse between a proposed exchange being approved and when the exchange can be executed. Should give sufficient time for Governance to veto an approved exchange.
```

Comment:
This would be cool, it'd definitely avoid issues if the governance parameters are made longer.

Something I was thinking would be cool is if governance had the ability to assume the role of the approver if it desires. It could set the `approver` to the GovernanceProxy address and then set `exchangeWaitPeriodSeconds` to zero (to avoid an exchange requiring basically 2 periods of a governance vote to be executed). Do you think that's compelling enough a reason to not go with the sum of governance parameters?